sdk/keyvault/azsecrets/client.go

```diff
@@ -271,11 +271,8 @@ func (s *startDeleteSecretPoller) Poll(ctx context.Context) (*http.Response, err
 		// Service recognizes DeletedSecret, operation is done
 		s.lastResponse = resp
 		return resp.RawResponse, nil
-	} else if err != nil {
```

Comment:
Why was this removed?  Looks like the error is now swallowed.

Reply:
The aboe error checked if `err == nil` so the return statement after the else was deadcode. I removed the else statement. I need to add a check to make sure the error is because of an expected error (this should return a 409 does not exist while polling), thanks for catching that.

Reply:
Fixed